website/src/pages/tech-specs.md

```diff
@@ -0,0 +1,371 @@
+
+
+
+## Storage Format
+
+### Layout Hierarchy
+
+At a high level, Hudi organizes data into a high level directory structure under the base path (root directory for the Hudi table). The directory structure is based on coarse-grained partitioning values set for the dataset. Non-partitioned data sets store all the data files under the base path. Hudi storage format has a special reserved *.hoodie* directory under the base path that is used to store transaction logs and metadata.
+
+```
+/data/hudi_trips/ 					<== BASE PATH
+├── .hoodie/      					<== META BASE PATH
+│   └── metadata/ 
+├── americas/
+│   ├── brazil/
+│   │   └── sao_paulo/      <== PARTITIONED DIRECTORY 
+│   │       ├── <data_files>
+│   └── united_states/
+│       └── san_francisco/
+│           ├── <data_files>
+└── asia/
+    └── india/
+        └── chennai/
+            ├── <data_files>
+```
+
+Hudi storage format offers two table types offering different trade-offs between ingest and query performance and the data files are stored differently based on the chosen table type. 
+
+| Table Type    | Trade-off                                                    |
+| ------------- | ------------------------------------------------------------ |
+| Copy on Write | Optimized for read performance and ideal for slow changing datasets |
+| Merge-on-read | Optimized to balance the write and read performance and ideal for frequently changing datasets |
+
+
+
+### Data Model
+
+Within each partition, data is organized into key-value model. Every row is uniquely identified with a row key. To write a row into Hudi dataset, each row must specify the following user fields 
+
+| User fields                 | Description                                                  |
+| --------------------------- | ------------------------------------------------------------ |
+| Partitioning key [Optional] | Value of this field defines the directory hierarchy within the table base path. This essentially provides an hierarchy isolation for managing data and related metadata |
+| Row key(s)                     | Record keys uniquely identify a record/row within each partition if partitioning is enabled |
+| Ordering field(s)           | Hudi guarantees the uniqueness constraint of row key and the conflict resolution configuration manages strategies on how to disambiguate when multiple records with the same keys are to be merged into the dataset. The resolution logic can be based on an ordering field or can be custom, specific to the dataset. To ensure consistent behaviour dealing with duplicate records, the resolution logic should be commutative and idempotent |
+
+**Hudi metadata fields**
+
+Hudi format stores the user fields along with the row merged along with transactional metadata fields. These fields are encoded in the data-file format and available in the table schema. 
+
+| Hudi meta-fields             | Description                                                  |
```

Comment:
there is also `_hoodie_commit_seqno` and optional `_hoodie_operation`

Reply:
_hoodie_commit_seqno is not relevant to the spec as of now I think. 
_hoodie_operation is an experimental cdc - I think we need to introduce CDC as a concept and how to efficiently do that over this spec. Leaving this as of now - as few more things are in flux at this point.